lib/ClangImporter/ImportDecl.cpp

```diff
@@ -4535,18 +4535,45 @@ namespace {
     }
 
     template <typename T, typename U>
-    T *resolveSwiftDeclImpl(const U *decl, Identifier name, ModuleDecl *overlay) {
+    T *resolveSwiftDeclImpl(const U *decl, Identifier name, bool hasCustomName,
+                            ModuleDecl *overlay) {
       const auto &languageVersion =
           Impl.SwiftContext.LangOpts.EffectiveLanguageVersion;
 
+      // None = not enough information
+      // true/false = definitive answer
+      auto isMatch = [&](const T *singleResult) -> Optional<bool> {
+        const DeclAttributes &attrs = singleResult->getAttrs();
+
+        // Skip versioned variants.
+        if (attrs.isUnavailableInSwiftVersion(languageVersion))
+          return false;
+
+        // If Clang decl has a custom Swift name, then `name` is that name.
+        if (hasCustomName)
+          return None;
```

Comment:
Can you explain more what this special case is about?

Reply:
This is in order to bypass comparing the custom Objective-C name with the known custom Swift name.